hcmanager/classification_filter.go

```diff
@@ -0,0 +1,264 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hcmanager
+
+import (
+	hwcc "hardware-classification-controller/api/v1alpha1"
+
+	bmh "github.com/metal3-io/baremetal-operator/pkg/apis/metal3/v1alpha1"
+)
+
+// MinMaxComparison it will compare the minimum and maximum comparison based on the value provided by the user and check for the valid host
+func (mgr HardwareClassificationManager) MinMaxComparison(ProfileName string, validatedHost map[string]map[string]interface{}, expectedHardwareprofile hwcc.HardwareCharacteristics) []string {
+
+	var comparedHost []string
+
+	for hostname, details := range validatedHost {
+		isHostValid := true
+		for _, value := range details {
+			isValid := false
+			cpu, CPUOK := value.(bmh.CPU)
+			if CPUOK {
+				if checkCPUCount(mgr, cpu, expectedHardwareprofile.Cpu) {
+					isValid = true
+				}
+			}
+
+			ram, RAMOK := value.(int64)
+			if RAMOK {
+				if checkRAM(mgr, ram, expectedHardwareprofile.Ram) {
+					isValid = true
+				}
+			}
+
+			nics, NICSOK := value.(int)
+			if NICSOK {
+				if checkNICS(mgr, nics, expectedHardwareprofile.Nic) {
+					isValid = true
+				}
+			}
+
+			disk, DISKOK := value.([]bmh.Storage)
+			if DISKOK {
+				if checkDiskDetails(mgr, disk, expectedHardwareprofile.Disk) {
+					isValid = true
+				}
+			}
+
+			if !isValid {
+				isHostValid = false
+				break
+			}
+
+		}
+
+		if isHostValid {
+			comparedHost = append(comparedHost, hostname)
+			mgr.Log.Info(hostname, " Matches profile ", ProfileName)
+
+		} else {
+			mgr.Log.Info(hostname, " Did not matches profile ", ProfileName)
+		}
+
+	}
+
+	return comparedHost
+
+}
+
+//checkCPUCount this function checks the CPU details for both min and max parameters
+func checkCPUCount(mgr HardwareClassificationManager, cpu bmh.CPU, expectedCPU *hwcc.Cpu) bool {
+
+	if (expectedCPU.MaximumCount > 0) && (expectedCPU.MinimumCount > 0) {
```

Comment:
we don't need braces here, 
It's ok to put like 
if expectedCPU.MaximumCount > 0 && expectedCPU.MinimumCount > 0

Reply:
Done